custom-email-module/src/main/java/org/wso2/carbon/custom/email/HTMLEmailSendingModule.java

```diff
@@ -0,0 +1,177 @@
+                        configContext.getAxisConfiguration().getTransportOut(EmailConstants.MAIL_TO);
+            }
+
+            String smtpHost =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_HOST).getValue().toString();
+            String smtpPort =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_PORT).getValue().toString();
+            String smtpStarttls =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_STARTTLS).getValue().toString();
+            String smtpAuth =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_AUTH).getValue().toString();
+            final String smtpUser = transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_USER).getValue()
+                    .toString();
+            final String smtpPassword =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_PASSWORD).getValue().toString();
+            String smtpFrom =
+                    transportOutDescription.getParameter(EmailConstants.MAIL_SMTP_FROM).getValue().toString();
+            String contentType;
+            if (transportOutDescription.getParameter(EmailConstants.MAIL_CONTENT_TYPE) != null) {
+                contentType =
+                        transportOutDescription.getParameter(EmailConstants.MAIL_CONTENT_TYPE).getValue().toString();
+            } else {
+                contentType = "text/plain";
+            }
+
+            Properties properties = new Properties();
+            properties.put(EmailConstants.MAIL_SMTP_HOST, smtpHost);
+            properties.put(EmailConstants.MAIL_SMTP_PORT, smtpPort);
+            properties.put(EmailConstants.MAIL_SMTP_STARTTLS, smtpStarttls);
+            properties.put(EmailConstants.MAIL_SMTP_AUTH, smtpAuth);
+
+            Session session = Session.getInstance(properties,
+                    new Authenticator() {
+                        protected PasswordAuthentication getPasswordAuthentication() {
+                            return new PasswordAuthentication(smtpUser, smtpPassword);
+                        }
+                    });
+
+            Message message = new MimeMessage(session);
+            Address[] addresses = InternetAddress.parse(notification.getSendTo());
+            message.setFrom(new InternetAddress(smtpFrom));
+            message.setRecipients(Message.RecipientType.TO, addresses);
+            message.setSubject(notification.getSubject());
+            StringBuilder contents = new StringBuilder();
+            contents.append(notification.getBody())
+                    .append(System.getProperty("line.separator"))
+                    .append(System.getProperty("line.separator"))
+                    .append(notification.getFooter());
+            message.setContent(contents.toString(), contentType);
+            Thread.currentThread().setContextClassLoader(MailUtils.class.getClassLoader());
```

Comment:
What is the use of setting the current thread? If this is actually needed, we need to reset this at the end of the function.